pkg/executor/slow_query.go

```diff
@@ -523,32 +522,94 @@ func getLineIndex(offset offset, index int) int {
 	return fileLine
 }
 
-// kvSplitRegex: it was just for split "field: value field: value..."
-var kvSplitRegex = regexp.MustCompile(`\w+: `)
+// findMatchedRightBracket returns the rightBracket index which matchs line[leftBracketIdx]
+// leftBracketIdx should be valid string index for line
+// Returns -1 if invalid inputs are given
+func findMatchedRightBracket(line string, leftBracketIdx int) int {
+	leftBracket := line[leftBracketIdx]
+	rightBracket := byte('}')
+	if leftBracket == '[' {
+		rightBracket = ']'
+	} else if leftBracket != '{' {
+		return -1
+	}
+	current := leftBracketIdx
+	leftBracketCnt := 0
+	for current < len(line) {
+		b := line[current]
+		if b == leftBracket {
+			leftBracketCnt++
+			current++
+		} else if b == rightBracket {
+			leftBracketCnt--
+			if leftBracketCnt > 0 {
+				current++
+			} else if leftBracketCnt == 0 {
+				return current
+			} else {
+				return -1
+			}
+		} else {
+			current++
+		}
+	}
+	return -1
+}
+
+func isLetterOrNumeric(b byte) bool {
+	return ('A' <= b && b <= 'Z') || ('a' <= b && b <= 'z') || ('0' <= b && b <= '9')
+}
 
 // splitByColon split a line like "field: value field: value..."
+// Note:
+// 1. Both field and value string contain only ANSI characters
```

Comment:
That is too restricted. E.g. tidb_redact_log may output non-ansi characters later sometime.
I am thinking if we could just use `strings.Index(a, b)` to mimic the original `regexp`.

Reply:
I see, this rule can be removed for current algorithm, I'll change it.
BTW, splitByColon is not the root entry for slow log parsing, for logs that may contain non-ansi characters we can handle it separately if we doneed it.

Reply:
Checked that whitespace character can't be a part of a valid non-whitespace utf8 character. Thus value string can contain non ascii characters except ' '.  And whitespace can be contained inside "{}"/"[]" in value string. For field string, first character should be letters or digits.